solr/solrj/src/java/org/apache/solr/common/util/SimpleOrderedMap.java

```diff
@@ -86,6 +91,79 @@ public static SimpleOrderedMap<Object> of() {
    * @return SimpleOrderedMap containing one key-value pair
    */
   public static <T> SimpleOrderedMap<T> of(String name, T val) {
-    return new SimpleOrderedMap<>(List.of(name, val));
+    return new SimpleOrderedMap<T>(List.of(name, val));
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return nvPairs.isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(final Object key) {
+    return super.get((String) key) != null;
```

Comment:
the value can actually be null in a NamedList, so this is inaccurate.  Use indexOf